### ISSUE TITLE
Can we define a stopping criterion for the tracker?

### ISSUE THREAD

**Issue by ekcheng** [davheld/GOTURN#64]
Based on this quote from the paper, do I understand correctly that GOTURN will track throughout the entire video regardless of whether the track has wandered off onto another object or whether the original object being tracked has left the scene?

> We continue to re-crop and feed pairs of frames into our network for the remainder
of the video, and our network will track the movement of the target object
throughout the entire video sequence.

If so, is there any mechanism for determining when a track has "wandered off"? Any score for the quality of the bounding box or the quality of the track that can be thresholded to prevent wandering tracks?

This is probably related to #60.

Thanks!

**Comment by davheld** [comment]
Not at the moment, sorry.  Feel free to make an extension that implements
this functionality!

**Comment by ekcheng** [comment]
Thanks, @davheld! Love to take a crack at implementing this functionality. Can you point me in the right general direction for how you'd approach a stopping algorithm?

I can think of two approaches off the top of my head. First, "bolt on" an independent metric of some sort that calculates the similarity of the object in the current frame v. the object you started tracking. If what you are now tracking has changed too much according to this metric, just stop the track.

Second, attack the problem during the training process. Instead of just training with ground truth boxes that are always on the object of interest, let the boxes track off the objects in the training set and also truth whether the box is on the object or not. Then update the CNN architecture to output an additional number between [0,1] which represents the probability that the box is still on the object.

Thoughts?
